src/model.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+from market_data import MarketData
+from enums import *
+from volgrid import VolGrid
+import numpy as np
+from abc import ABC, abstractmethod
+
+class MarketModel(ABC):
+    def __init__(self, und: Stock):
+        self._und = und
+        self._interest_rate = MarketData.get_interest_rate()
+        self._initial_spot = MarketData.get_initial_spot(self._und)
+        self._volgrid = MarketData.get_volgrid(self._und)
+
+    def get_rate(self) -> float:
+        return self._interest_rate
+
+    def get_initial_spot(self) -> float:
+        return self._initial_spot
+
+    def get_volgrid(self) -> VolGrid:
+        return self._volgrid
+
+    def get_df(self, t: float) -> float:
+        return np.exp(-self.interest_rate * t)
+
+    @abstractmethod
+    def get_simulated_spot(self, t: float, Z: float) -> float:
+        pass
+
+
+class BSVolModel(MarketModel):
+    def __init__(self, und: Stock):
+        super().__init__(und)
+
+    def get_vol(self) -> float:
+        return self.get_volgrid.get_vol(1.0, 1.0)
+
+    def get_simulated_spot(self, t: float, Z: float) -> float:
+        return self._initial_spot * np.exp(
```

Comment:
I think we agreed that we'll do the simulation and the pricing in percentage terms i.e. assuming S(0) = 1.0. Then if user wants, they can display the results in units or in percentages.

Reply:
I see. self._initial_spot has been removed

Reply:
Not sure if we need get_simulated_spot here, the simulation is handeled in the numerical method module

Reply:
This is open for debate. I think the model should own the diffusion and the solution of the eq. that can be used for simulation. Then, the numerical method should generate the random numbers and provide them to this function as input. In case of changing the diffusion, ideally the numerical method should not be touched. This is not true, if we hardcode the equation there.